flink-clients/src/test/java/org/apache/flink/client/deployment/application/ClassPathPackagedProgramRetrieverTest.java

```diff
@@ -315,6 +327,24 @@ public void testRetrieveCorrectUserClasspathsWithSpecifiedEntryClass()
                 containsInAnyOrder(expectedURLs.stream().map(URL::toString).toArray()));
     }
 
+    @Test
```

Comment:
I think the new introduced test does not cover the merging with usrlib and `pipeline.classpaths`. What do you think about making the test like this?

```
    @Test
    public void testRetrieveCorrectUserClasspathsWithPipelineClasspaths()
            throws IOException, FlinkException, ProgramInvocationException {
        final Configuration configuration = new Configuration();
        final List<String> pipelineJars =
                Arrays.asList("file:///path/of/p1.jar", "http://path/of/p2.jar");
        configuration.set(PipelineOptions.CLASSPATHS, pipelineJars);

        final ClassPathPackagedProgramRetriever retrieverUnderTest =
                ClassPathPackagedProgramRetriever.newBuilder(PROGRAM_ARGUMENTS)
                        .setUserLibDirectory(userDirHasEntryClass)
                        .setJarFile(TestJob.getTestJobJar())
                        .setConfiguration(configuration)
                        .build();
        final JobGraph jobGraph = retrieveJobGraph(retrieverUnderTest, new Configuration());

        final List<URL> expectedMergedURLs = new ArrayList<>(expectedURLs);
        expectedMergedURLs.addAll(
                pipelineJars.stream()
                        .map(FunctionUtils.uncheckedFunction(URL::new))
                        .collect(Collectors.toList()));
        assertThat(
                jobGraph.getClasspaths().stream().map(URL::toString).collect(Collectors.toList()),
                containsInAnyOrder(expectedMergedURLs.stream().map(URL::toString).toArray()));
    }
```

Reply:
yes, it should be better.